flow/directives.py

```diff
@@ -0,0 +1,315 @@
+from collections.abc import MutableMapping
+import sys
+
+
+class _DirectivesItem:
+    """The representation of a single directive.
+
+    Logic for validation of values when setting, defaults, and the ability for
+    directives to interspect before returning a directive are supported. This is
```

Comment:
"ability for directives to interspect": I'm not sure what is meant by this. Perhaps you mean "introspect" or "inspect" but that requires something else in the sentence (a direct object, I believe) to be introspected / inspected.